### PR TITLE
Formations

### PR DESCRIPTION
**Pull request by Nathaaniel**

### CODE REVIEW

File: app/views/matches/show.html.erb [hunk 1]
```diff
@@ -1,46 +1,125 @@
...
-        <td><%= @home_players.map{ |pl| pl.gameweek_points[@match.gameweek] }.sum + @home_captain.gameweek_points[@match.gameweek] %></td>
-        <td></td>
-        <td><%= @away_players.map{ |pl| pl.gameweek_points[@match.gameweek] }.sum + @away_captain.gameweek_points[@match.gameweek]  %></td>
-        <td></td>
-        </tr>
-      <% end %>
-    </tbody>
-  </table>
+
+
```
Comment (kimjongxav) [inline app/views/matches/show.html.erb]: you never want to leave more than one new line anywhere in any file

this file in general, can you sort the nesting and formatting out? it makes it so much easier to read

File: app/views/matches/show.html.erb [hunk 2]
```diff
@@ -1,46 +1,125 @@
...
+<% def position(pos, team) %>
+    <% return team.select{|k| pos==k[1]}.map{|k| [k[0],k[2], k[3], k[4]]} %>
```
Comment (kimjongxav) [inline app/views/matches/show.html.erb]: ruby returns implicitly - every method returns the last line, so you never need to return on the last line

File: app/views/matches/show.html.erb [hunk 3]
```diff
@@ -1,46 +1,125 @@
...
-        <td></td>
-        <td><%= @away_players.map{ |pl| pl.gameweek_points[@match.gameweek] }.sum + @away_captain.gameweek_points[@match.gameweek]  %></td>
-        <td></td>
-        </tr>
-      <% end %>
-    </tbody>
-  </table>
+
+
+<% def position(pos, team) %>
```
Comment (kimjongxav) [inline app/views/matches/show.html.erb]: your suspicion was right and you definitely don't want to be writing methods in .html.erb files

try moving the methods to `app/models/match.rb`

File: app/assets/stylesheets/shared/matches.css [hunk 4]
```diff
@@ -0,0 +1,60 @@
+.image {
```
Comment (kimjongxav) [inline app/assets/stylesheets/shared/matches.css]: this stuff should all be in matches.scss

File: app/views/matches/show.html.erb [hunk 5]
```diff
@@ -1,46 +1,125 @@
...
+  </div>
+  <div class="positioning">
+  <% fwdh = position("FWD", homeTeam) %>
+  <% showImages(fwdh) %>
+  </div>
+  </div>
+</div>
+
+
+<div class="away">
```
Comment (kimjongxav) [inline app/views/matches/show.html.erb]: this away class is exactly the same as the home class - you don't need both - calling it something more general

File: app/views/matches/show.html.erb [hunk 6]
```diff
@@ -1,46 +1,125 @@
...
+
+  
+  <div class="team-score">
```
Comment (kimjongxav) [inline app/views/matches/show.html.erb]: this whole away section is just a repeat of the home one

we want our code to be [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself)

try to think of a way to reuse the code above without typing it all out again like it is now

File: db/seeds.rb [hunk 7]
```diff
@@ -65,7 +65,7 @@ def position(type)
   'Xavier Lacey',
 ]
 initials = %w[AD AD2 AL CT DW GM HT JK JT KW NM PB PL XL]
-email = %w[
+emails = %w[
```
Comment (kimjongxav) [inline db/seeds.rb]: 👍

File: db/seeds.rb [hunk 8]
```diff
@@ -181,28 +182,30 @@ def position(type)
     :common_name => surname,
     :position => position,
     :premier_league_team_id => player['team'],
+    :picture => picture,
     :gameweek_points => { 1 => 0 },
+
```
Comment (kimjongxav) [inline db/seeds.rb]: ✂️

File: db/seeds.rb [hunk 9]
```diff
@@ -181,28 +182,30 @@ def position(type)
...
-  '19' => [262, 206, 337, 144, 28, 265, 390, 292, 187, 36, 68], # KW
-  '20' => [14, 202, 203, 515, 226, 193, 15, 215, 239, 457, 11], # NM
-  '21' => [189, 252, 3, 159, 191, 266, 320, 368, 90, 466, 460], # PB
-  '22' => [366, 183, 332, 186, 218, 116, 220, 414, 415, 487, 167], # PL
-  '23' => [198, 331, 106, 405, 344, 176, 199, 219, 278, 211, 437], # XL
-  '24' => [94, 182, 381, 258, 325, 442, 488, 525,283, 338, 428], # AD2
-  '25' => [340, 105, 2, 160, 150, 238, 342, 134, 133, 188, 313], # AL
-  '26' => [411, 330, 179, 59, 75, 448, 392, 469, 50, 233, 44], # GM
-  '27' => [168, 42, 163, 39, 10, 24, 393, 244, 347, 365, 175], # HT
+  '1' => [93, 141, 401, 182, 518, 192, 431, 151, 217, 67, 409], # AD1
```
Comment (kimjongxav) [inline db/seeds.rb]: why did you change all these numbers?

File: db/schema.rb [hunk 10]
```diff
@@ -66,6 +66,7 @@
     t.string "full_name"
     t.string "common_name"
     t.string "position"
+    t.string "picture"
```
Comment (kimjongxav) [inline db/schema.rb]: this isn't how you add fields to the db, you need to do a [migration](https://edgeguides.rubyonrails.org/active_record_migrations.html#creating-a-migration)

File: Gemfile.lock [hunk 11]
```diff
@@ -1,60 +1,60 @@
 GEM
```
[the remark this answers is not in the document]
Comment (Nathaaniel) [reply Gemfile.lock]: I didn't realise I did. How do I un-updated them?

File: app/assets/stylesheets/shared/matches.css [hunk 12]
```diff
@@ -0,0 +1,60 @@
...
+  
+
```
[the remark this answers is not in the document]
Comment (Nathaaniel) [reply app/assets/stylesheets/shared/matches.css]: Done

File: app/assets/stylesheets/shared/nice-table.scss [hunk 13]
```diff
@@ -1,3 +1,66 @@
+.image {
```
[the remark this answers is not in the document]
Comment (Nathaaniel) [reply app/assets/stylesheets/shared/nice-table.scss]: It does? I can't see any shared styling...
It was previously in there but then I removed it

File: app/views/matches/show.html.erb (hunk 1 above)
Comment (Nathaaniel) [reply app/views/matches/show.html.erb]: Done

File: app/views/matches/show.html.erb (hunk 6 above)
Comment (Nathaaniel) [reply app/views/matches/show.html.erb]: Done

File: db/seeds.rb (hunk 9 above)
Comment (Nathaaniel) [reply db/seeds.rb]: It was giving me the wrong (last year's) teams, I think it must be different in live?

File: db/schema.rb (hunk 10 above)
Comment (Nathaaniel) [reply db/schema.rb]: I've added       t.string :picture to the first migration file, is that right?
